src/utils/loose-equal.js

```diff
@@ -16,44 +19,27 @@ function isObject(obj) {
  * Returns boolean true or false
  */
 function looseEqual(a, b) {
-  if (a === b) return true
-  const isObjectA = isObject(a)
-  const isObjectB = isObject(b)
-  if (isObjectA && isObjectB) {
-    try {
-      const isArrayA = isArray(a)
-      const isArrayB = isArray(b)
-      if (isArrayA && isArrayB) {
-        return (
-          a.length === b.length &&
-          a.every((e, i) => {
-            return looseEqual(e, b[i])
-          })
-        )
-      } else if (a instanceof Date && b instanceof Date) {
-        return a.getTime() === b.getTime()
-      } else if (!isArrayA && !isArrayB) {
-        const keysA = keys(a)
-        const keysB = keys(b)
-        return (
-          keysA.length === keysB.length &&
-          keysA.every(key => {
-            return looseEqual(a[key], b[key])
-          })
-        )
-      } else {
-        /* istanbul ignore next */
+  if (a === b) {
+    return true
+  }
+  if (isObject(a) && isObject(b)) {
+    // Using for loop over `Object.keys()` here since some class
+    // keys are not handled correctly otherwise
+    for (const key in a) {
```

Comment:
Should we not do a quick comparison of the number of keys in each object for speed?

i.e. the original Vue version did a quick check on `Object.keys(a).length !== Object.keys(b).length` and returned `false` if not equal.

Reply:
i.e. if we had the following two objects with the same shape, esxcept for one additional key in the b object:

```js
const a = { a: 1, b: 2, c: 3 }
const b = { a: 1, b: 2, c: 3, d: 4 }
```

Reply:
We can do that before the for loop for sure.